src/main/java/com/develhope/spring/vehicles/controller/VehicleController.java

```diff
@@ -13,13 +15,44 @@
 @RestController
 @RequestMapping("/vehicle")
 public class VehicleController {
+
     @Autowired
     private VehicleService vehicleService;
 
     @PostMapping
     public ResponseEntity<Vehicle> createVehicle(@RequestBody Vehicle vehicle) {
-        Vehicle vehicleAdded = vehicleService.createVehicle(vehicle);
-        return ResponseEntity.ok(vehicleAdded);
+        try {
+            Vehicle vehicleAdded = vehicleService.createVehicle(vehicle);
+            return ResponseEntity.ok(vehicleAdded);
+        } catch (IllegalArgumentException e) {
+            return ResponseEntity.badRequest().build();
+        }
+    }
+
+    @PostMapping("/shop")
```

Comment:
In questa post manca effettivamente l'id dell'utente effettuante la richiesta, inoltre non sono sicuro che vada effettuata una post ma per ora va bene così